spec/01-unit/06-timestamp_spec.lua

```diff
@@ -55,4 +56,29 @@ describe("Timestamp", function()
     assert.are.same(timestamps_one, timestamps_two)
   end)
 
+  it("should provide correct timestamp values", function()
```

Comment:
Can I point out that this test is very confusing and hard to understand at a glance?

My 2 cents:
- split it in at least 2 tests: ms and s precision. This is also beneficial in that it makes the tests more atomic, because we can maybe fail the `ms` or `s` only behavior part of the function, and when that happens, we still want the other test to pass, pointing out that only part of the function's behavior is broken. This is mostly about good practice for unit testing, I believe.
- add more line-jumps comments explaining the process here and the "magical numbers" being used (we get that it is to jump to different timing precisions but they are all squashed together and it's quite hard to read).